pkg/kubelet/cm/dra/checkpoint/checkpoint.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package checkpoint
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/kubelet/checkpointmanager/checksum"
+	state "k8s.io/kubernetes/pkg/kubelet/cm/dra/state/v1"
+)
+
+const (
+	CheckpointAPIGroup   = "checkpoint.dra.kubelet.k8s.io"
+	CheckpointKind       = "DRACheckpoint"
+	CheckpointAPIVersion = CheckpointAPIGroup + "/" + state.Version
+)
+
+// Checkpoint represents a structure to store DRA checkpoint data
+type Checkpoint struct {
+	// Data is a JSON serialized checkpoint data
+	// See state.CheckpointData for the details
+	Data string
+	// Checksum is a checksum of Data
+	Checksum checksum.Checksum
+}
+
+type CheckpointData struct {
```

Comment:
From an architectural perspective it looks like this struct and the API group and version should be in the v1 package.

Then this file could be completely agnostic of the underlying type by using a scheme. But I understand that this is too much for now and doesn't pay off unless we actually need conversion, so fine with me.

Reply:
I had ed move everything into an unversioned package for now (but keep the APIGroup and v1 versioning in the type). That way we can figure out the right way to move things around for 1.32 without overcomplicating things for now.